### ISSUE TITLE
FeatureRequest: NavigationMesh as Polygons to define BotMovementArea

### ISSUE THREAD

**Issue by cabrinhaFx** [finndev/PokeBuddy#176]
Hey,

I`ve noticed, that the bot is going very often out of my preferred area ... so I loose a ton of experience. So my request would be ... to be able to draw some polygons on top of the map, where the bot is allowed to run around and the rest gets kinda marked as forbidden area, where the bot won`t take a step in. So the whole logic is bound to your drawn polygons.

I attached an example ... green for allowed navigation, red for not allowed one. User should define the green one and the red one get`s automatically generated.

![google_polygon1](https://cloud.githubusercontent.com/assets/13951053/17346041/4ac34ef4-5909-11e6-9c50-685ec802f4e1.jpg)

**Comment by lordofthereef** [comment]
I'll second this. The bot does "wander off" and then tends to stay in it's "own preferred" area. It hasn't effected my overall exp, but at this point I'd rather be pinpointing specific known nests, which I cannot really do .

**Comment by yuneedmyname** [comment]
Agreed. I don't care how it's implemented, but it would be nice. I went to bed last night with coordinates set between a Bulbasaur nest and a Magmar nest (with only about 4 blocks between them), and woke up to find that the bot had gone all over Prague. I'm not too upset, though... it managed to find our Scyther nest :)

**Comment by lordofthereef** [comment]
In starting to wonder if all your movement speed has anything to do with how far your not wanders. I'm in a dense area and I've set my speed to 11 (I believe it's a manageable walk speed of still a bit fast) and don't tend to wander very far at all (maybe 1/5 mile). 

This isn't to say this feature wouldn't be great, just curious what all your settings are.

**Comment by cabrinhaFx** [comment]
I´m running around with 21km/h in central park, there is a huge amount of PokeStops, so the Bot sometimes takes the wrong decission and moves the wrong way (xp-drop from around 60k to 30k). I think the bot behaves like: Go to the nearest PokeStop and catch every pokemon on your way, so his routine is kinda complete random in direction. I don`t think it has anything to do with the movement-speed, it`s just about ... where you are farming with your bot ... a lot of PokeStops crowded, a lot of random decissions and a high possibility for the bot to go the wrong way.

Another possibility and an easier implementation would be, to be able to mark the pokestops by clicking, which can be target by the bot as the nearest PokeStop, the rest are just gonna be ignored ... so the Bot should wander around your specified, marked PokeStops ... on dense areas ... you got a lot to mark than, but maybe with some kind of profile-system, you could easily exchange your marked spots with the community.

**Comment by Vanethir** [comment]
First feature added to list. :) Prefer the GPX feature you mentioned in your second post though. ;)
Either by clicking on a Pokestop or just by adding a list of coordinates.
